index.js

```diff
@@ -300,7 +300,7 @@ Router.prototype.find = function find (method, path) {
     i = pathLen < prefixLen ? pathLen : prefixLen
     while (len < i && path[len] === prefix[len]) len++
 
-    if (len === prefixLen) {
+    if (len > 0) {
```

Comment:
`if (len === prefixLen && len > 0)`?

Reply:
Nope. The problem is the `len === prefixLen` part.

Reply:
I think maybe `len === prefixLen` is correct and something else needs to change.